src/test/groovy/com/palantir/gradle/gitversion/GitVersionPluginTests.groovy

```diff
@@ -611,6 +611,98 @@ class GitVersionPluginTests extends Specification {
         buildResult.output.contains(":printVersion\n1.0.0\n")
     }
 
+    def 'test build service cache results with the same work tree' () {
+        given:
+        buildFile << '''
+            plugins {
+                id 'com.palantir.git-version'
+            }
+            version gitVersion()
+            subprojects {
+                apply plugin: 'com.palantir.git-version'
+                version gitVersion()
+            }
+        '''.stripIndent()
+        File subModuleDir1 = Files.createDirectories(projectDir.toPath().resolve('submodule1')).toFile()
+        File subModuleBuildFile1 = new File(subModuleDir1, 'build.gradle')
+        subModuleBuildFile1.createNewFile()
+        File subModuleDir2 = Files.createDirectories(projectDir.toPath().resolve('submodule2')).toFile()
+        File subModuleBuildFile2 = new File(subModuleDir2, 'build.gradle')
+        subModuleBuildFile2.createNewFile()
+        subModuleBuildFile2 << '''
+            printVersion {
+                doFirst{
+                    exec {
+                        commandLine 'git', '-c', "user.email='email@example.com'", '-c','user.name=name', 'tag', '-a', '2.0.0', '-m', '2.0.0'
+                    }
+                }
+            }
+        '''.stripIndent()
+        gitIgnoreFile << 'build'
+        settingsFile << '''
+            include 'submodule1'
+            include 'submodule2'
+        '''.stripIndent()
+        println(projectDir.toString())
+        Git git = Git.init().setDirectory(projectDir).call()
+        git.add().addFilepattern('.').call()
+        git.commit().setMessage('initial commit').call()
+        git.tag().setAnnotated(true).setName('1.0.0').setMessage('1.0.0').call()
+        BuildResult buildResult1 = with(':submodule1:printVersion').build()
+
+        when:
+        BuildResult buildResult2 = with(':submodule2:printVersion').build()
+
+        then:
+        buildResult1.output.contains(":printVersion\n1.0.0\n")
+        buildResult2.output.contains(":printVersion\n1.0.0\n")
+    }
```

Comment:
I'd like to explain about this test case a little bit. The build file of `submodule2` makes sure that before we are calling `:submodule2:printVersion`, we change the latest tag to be `2.0.0` in the work tree. However, since the build service will always be registered at configuration time by root project(at a time that this git command isn't executed) and the call to `printVersion` on `submodule2` is always cached, the result of `:submodule2:printVersion` would be `1.0.0`. 

The thing that I am worried about is, even though this test case showcases that caching is indeed successful, does it also imply a potential bug/issue with the logic? Suppose we have a build file like `subModuleBuildFile2` where the git status of the repo changes during execution time. Then the correctness of the plugin cannot be guaranteed unless we add a bunch of expensive operations to check if the tags have been modified.

Reply:
I think assuming that the underlying state of the git tree does not change during a Gradle execution is a reasonable. The thing we want to make sure is we do not cache the version details between Gradle executions.

Reply:
Cool. I can delete the new test cases I wrote.